### PR TITLE
Add document explaining v1beta1 changes

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
I ended up adding a more general "CRD changelog" document, with the first entry explaining the v1alpha1 -> v1beta1 transition. We can put other, similar sections in here in the future, for example introducing new CRDs.

### CODE REVIEW

File: docs/crd-changelog.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# CRD Changelog
+
+This document explains major changes made in new CRD versions. It is intended to help users migrate and take
+advantage of the new features.
+
+## opentelemetry.io/v1beta1 OpenTelemetryCollector
```
Comment (pavolloffay) [inline docs/crd-changelog.md:6]: This is commonly written as OpenTelemetryCollector.opentelemetry.io/v1beta1

File: docs/crd-changelog.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+### Migration
+
+There is no need for any user action. The operator will automatically convert existing `v1alpha1` resources to `v1beta1`.
+
+### Structured Configuration
+
+The `Config` field containing the Collector configuration is a string in `v1alpha1`. This has some downsides:
```
Comment (pavolloffay) [inline docs/crd-changelog.md:14]: I would call out the limitation with kustomize and other tools

File: docs/crd-changelog.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp]
+          processors: [memory_limiter, batch]
+          exporters: [debug]
+```
+
+> [!NOTE]  
+> Empty maps, like `debug:` in the above configuration, should have an explicit value of `{}`.
```
Comment (pavolloffay) [inline docs/crd-changelog.md:88]: !! :)

File: docs/crd-changelog.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+1. Update any manifests you have stored outside the cluster, for example in your infrastructure git repository.
+2. Apply them, so they're all stored as `v1beta1`.
+3. Update the OpenTelemetryCollector CRD to only store `v1beta1`
+   ```bash
+   kubectl patch customresourcedefinitions opentelemetrycollectors.opentelemetry.io  \
+     --subresource='status' \
+     --type='merge' \
+     -p '{"status":{"storedVersions":["v1beta1"]}}'
+   ```
+For a more thorough explanation of how and why this migration works, see the relevant [Kubernetes documentation][crd_migration guide].
```
Comment (jaronoff97) [inline docs/crd-changelog.md:24]: shouldn't there be a link for the kube documentation there?

Comment (jaronoff97) [reply docs/crd-changelog.md:24]: actually this may just be the github preview renderer not working

Comment (jaronoff97) [reply docs/crd-changelog.md:24]: no, nvm i think this should be []() syntax right?

File: docs/crd-changelog.md (hunk 2 above)
Comment (swiatekm-sumo) [reply docs/crd-changelog.md:35]: I called it out below, would you prefer it closer to the start of the section?

File: docs/crd-changelog.md (hunk 4 above)
Comment (swiatekm-sumo) [reply docs/crd-changelog.md]: Yeah, it was missing an underscore.

File: docs/crd-changelog.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+This document explains major changes made in new CRD versions. It is intended to help users migrate and take
+advantage of the new features.
+
+## OpenTelemetryCollector.opentelemetry.io/v1beta1 
+
+### Migration
+
+There is no need for any immediate user action. The operator will automatically convert existing `v1alpha1` resources to `v1beta1`.
+
+However, the conversion procedure will leave your resources stored as both versions. Support for `v1alpha1` will be removed in the future.
```
Comment (pavolloffay) [inline docs/crd-changelog.md:12]: This is not accurate, but rather misleading. 

>  the conversion procedure will leave your resources stored as both versions.

The custom resources stored as v1alpha1 are not automatically updated to v1beta1. The updated or newly written cutom resources will be stored as v1beta1 because it is the new storage version. 

I would also link https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version

File: docs/crd-changelog.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
+# CRD Changelog
+
+This document explains major changes made in new CRD versions. It is intended to help users migrate and take
+advantage of the new features.
+
+## OpenTelemetryCollector.opentelemetry.io/v1beta1 
+
+### Migration
+
+There is no need for any immediate user action. The operator will automatically convert existing `v1alpha1` resources to `v1beta1`.
```
Comment (pavolloffay) [inline docs/crd-changelog.md:10]: >  The operator will automatically convert existing `v1alpha1` resources to `v1beta1`.

I would rephrase this. Yes there is conversion webhook but for migration from v1alpha1 to v1beta1 an update on the CR must happen

File: docs/crd-changelog.md (hunk 5 above)
Comment (swiatekm-sumo) [reply docs/crd-changelog.md:12]: Ok, how about:

```
There is no need for any immediate user action. The operator will continue to support existing `v1alpha1` resources.

In addition, any newly applied `v1alpha1` resource will be converted to `v1beta1` and stored as both versions. The plan is to remove support for `v1alpha1` in a future version, so users should migrate promptly.

(detailed migration guide here)
```

**Review by jaronoff97 — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]
